src/pheval/cli_pheval_utils.py

```diff
@@ -27,3 +34,77 @@ def scramble_semsim():
 def scramble_phenopacket():
     """scramble_phenopacket"""
     print("running pheval_utils::scramble_phenopacket command")
+
+
+@click.command()
+@click.option(
+    "--left",
+    "-l",
+    required=True,
+    metavar="FILE",
+    help="Path of first file that will be compared",
+)
+@click.option(
+    "--right",
+    "-r",
+    required=True,
+    metavar="FILE",
+    help="Path of second file that will be compared",
+)
+@click.option(
+    "--join-columns",
+    "-j",
+    required=False,
+    multiple=True,
+    default=["GENE_SYMBOL"],
+    metavar="LIST",
+    help="""Represents the parameter that specifies which
+columns will be used to join left and right inputs.""",
+)
+@click.option(
+    "--comparison-columns",
+    "-c",
+    required=True,
+    multiple=True,
+    metavar="LIST",
+    help="""Represents the parameter that specifies which columns will be
+compared in left and right inputs.
+This comparison is just a value difference between those two files.
+e.g (left = 5; right = 2; diff = 3)""",
+)
+@click.option(
+    "--sort-columns",
+    "-s",
+    required=True,
+    multiple=True,
+    metavar="LIST",
+    default=["P-VALUE"],
+    help="""Represents the parameter that specifies which columns will
+be used to rank genes.
+- if sort column starts with "-", it will be interpreted as reverse order rank""",
+)
+@click.option(
+    "--output",
+    "-o",
+    required=True,
+    metavar="FILE",
+    help="Path to the result output",
+)
+def compare_semsim(
+    left: click.Path,
+    right: click.Path,
+    output: click.Path,
+    join_columns: List[str],
+    comparison_columns: List[str],
+    sort_columns: List[str],
+):
+    """compare_semsim"""
+    try:
+        dataframe = utils.calc_semsim(
```

Comment:
One very important design decision will be to only use LinkML conformant models when passing around results. I think in the long run, you should return a dictionary structure here, because `calc_semsim` will compute many different metrics.